tools/actions_local_runner.py

```diff
@@ -70,65 +78,169 @@ def find_changed_files() -> List[str]:
+        f"{sys.executable} tools/actions_local_runner.py --job 'mypy' --file .github/workflows/lint.yml --step 'Run mypy'",
+        redirect=False,
+    )
+    return passed
+
+
+async def run_step(
+    step: Dict[str, Any], job_name: str, files: Optional[List[str]], quiet: bool
+) -> bool:
     env = os.environ.copy()
     env["GITHUB_WORKSPACE"] = "/tmp"
-    if files is None:
-        env["LOCAL_FILES"] = ""
-    else:
-        env["LOCAL_FILES"] = " ".join(files)
     script = step["run"]
 
-    PASS = color(col.GREEN, '\N{check mark}')
-    FAIL = color(col.RED, 'x')
-
     if quiet:
         # TODO: Either lint that GHA scripts only use 'set -eux' or make this more
         # resilient
         script = script.replace("set -eux", "set -eu")
         script = re.sub(r"^time ", "", script, flags=re.MULTILINE)
     name = f'{job_name}: {step["name"]}'
 
-    def header(passed: bool) -> None:
-        icon = PASS if passed else FAIL
-        print(f"{icon} {color(col.BLUE, name)}")
+    passed, stderr, stdout = await shell_cmd(script, env=env)
+    print_results(name, passed, [stdout, stderr])
 
-    try:
-        proc = await asyncio.create_subprocess_shell(
-            script,
-            shell=True,
-            cwd=REPO_ROOT,
-            env=env,
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-            executable=shutil.which("bash"),
-        )
+    return passed
 
-        stdout_bytes, stderr_bytes = await proc.communicate()
 
-        header(passed=proc.returncode == 0)
-    except Exception as e:
-        header(passed=False)
-        print(e)
-        return False
-
-    stdout = stdout_bytes.decode().strip()
-    stderr = stderr_bytes.decode().strip()
-
-    if stderr != "":
-        print(stderr)
-    if stdout != "":
-        print(stdout)
-
-    return proc.returncode == 0
+async def run_steps(
+    steps: List[Dict[str, Any]], job_name: str, files: Optional[List[str]], quiet: bool
+) -> bool:
+    coros = [run_step(step, job_name, files, quiet) for step in steps]
+    return all(await asyncio.gather(*coros))
 
 
-async def run_steps(steps: List[Dict[str, Any]], job_name: str, files: Optional[List[str]], quiet: bool) -> None:
-    coros = [run_step(step, job_name, files, quiet) for step in steps]
-    await asyncio.gather(*coros)
+def relevant_changed_files(file_filters: Optional[List[str]]) -> Optional[List[str]]:
+    changed_files: Optional[List[str]] = None
+    try:
+        changed_files = find_changed_files()
+    except Exception:
+        # If the git commands failed for some reason, bail out and use the whole list
+        print(
+            "Could not query git for changed files, falling back to testing all files instead",
+            file=sys.stderr,
+        )
+        return None
 
+    if changed_files is None:
+        print(
+            "Did not find any changed files, falling back to testing all files instead",
+            file=sys.stderr,
+        )
+        return None
```

Comment:
this is kind of confusing:

1. if the Git operations successfullyl run and there aren't any changed files, why would `quicklint` then fallback to running on all files? it seems like it would make more sense for it to run on no files in that case, rather than all files
2. from looking at the source code for `find_changed_files`, I don't see a case in which it returns `None`; what exactly does this mean? it seems like if there are actually no changed files, it would return an empty list instead of `None` (and its return type is `List[str]` rather than `Optional[List[str]]`

seems like this block can/should just be removed:
```suggestion
```